### ISSUE TITLE
Focus is lost from Layer (modals) to underlying content once you tab out of Layer

### ISSUE THREAD

**Issue by eddieshowcase** [grommet/grommet#5509]
Using Layer component for modals, particularly with a form, I noticed what I believe to be a pretty bad accessibility issue that focus is not contained to the Layer. 

If you tab through elements in a Layer (modal) form the focus seems to exit the Layer and move back to the underlying content and can be lost. This will cause a user relying on keyboard navigation to get lost and or unable to close the Layer and proceed. 

This can be seen in the drawer example with form on website here: https://design-system.hpe.design/components/layer

### Expected Behavior

Tab focus via keyboard navigation does not stay contained in the Layer component.

### Actual Behavior

Tab focus leaves the Layer container and is then lost to underlying content.

### URL, screen shot, or Codepen exhibiting the issue

<img width="1610" alt="Screen Shot 2021-08-04 at 9 49 51 AM" src="https://user-images.githubusercontent.com/7239901/128213705-0647c284-4d28-4b4d-ad13-e32c068499f0.png">

### Steps to Reproduce

Open Layer (modal), start tabbing and watch where the focus goes.

### Your Environment

- Grommet version: "grommet": "^2.17.4",
- Browser Name and version: Chrome (latest)
- Operating System and version (desktop or mobile): Mac OS

**Comment by taysea** [comment]
Hey @eddieshowcase, the reason the focus is not trapped inside this Layer is because it was implemented as `modal={false}` which means that focus is not trapped.

Since behavior of Layer is correct here, I am going to close the issue. However, we can take this implementation question of "should this be a modal as opposed to non-modal?" in the design system context offline.

**Comment by eddieshowcase** [comment]
I see now, thanks! I didn't see the modal={false} prop option..... still learning all these components.
